File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        pal = ""
        pallen = 0
        n = len(s)

        for i in range(n):
            l,r = i,i
            while(l>=0 and r<n and s[l]==s[r]):
                if pallen < r-l+1:
                    pal = s[l:r+1]
                    pallen = r-l+1
                l-=1
                r+=1
            l-=1
            r+=1
            l,r = i,i+1
            while(l>=0 and r<n and s[l]==s[r]):
                if pallen < r-l+1:
                    pal = s[l:r+1]
                    pallen = r-l+1
                l-=1
                r+=1
            l-=1
            r+=1
        return pal

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.java
class Solution {
    public String longestPalindrome(String s) {
        int n = s.length();
        String pal = "";
        int pallen = 0;
        for(int i=0;i<n;i++)
        {
                int l=i,r=i;
                while(l>=0 && r<n && s.charAt(l)==s.charAt(r))
                {
                    if(r-l+1>pallen){
                        pal = s.substring(l,r+1);
                        pallen = r-l+1;
                    }
                    l-=1;
                    r+=1;
                }

                l=i; r=i+1;
                while(l>=0 && r<n && s.charAt(l)==s.charAt(r))
                {
                    if(r-l+1>pallen){
                        pal = s.substring(l,r+1);
                        pallen = r-l+1;
                    }
                    l-=1;
                    r+=1;
                }
        }
        return pal;
    }
}